src/renderer/fonts/caching_shaper.rs

```diff
@@ -175,11 +175,26 @@ impl CachingShaper {
 
     pub fn font_base_dimensions(&mut self) -> (u64, u64) {
         let (metrics, glyph_advance) = self.info();
-        let font_height =
-            (metrics.ascent + metrics.descent + metrics.leading).ceil() as u64 + self.linespace;
         let font_width = (glyph_advance + 0.5).floor() as u64;
+        let mut font_height = (metrics.ascent + metrics.descent + metrics.leading).ceil() as i64;
 
-        (font_width, font_height)
+        let font_height_threshold = 10;
```

Comment:
We should have a threshold in case the `linespace` value is smaller than the actual font-height.

I set it to `10`. On my screen this results in what you see on the screenshot below when setting something like `linespace=-100`:
![Screenshot_20230208_185525](https://user-images.githubusercontent.com/34311583/217615786-911e50cc-761b-4e1d-867d-96917a638596.png)

I'm very open to other suggestions on how to solve this.

Reply:
Hm, could you explain your reasoning behind the threshold? I don't really see the correlation between `linespace` being smaller than the font height and the threshold.

Reply:
I think instead of applying an opaque default, it might be more clear to ignore the `linespace` setting in that case and throw an error, telling that the `linespace` exceeds the font height. Probably something similar if a font hasn't been found would be a good option, too.